.github/workflows/ci.yml

```diff
@@ -98,20 +98,17 @@ jobs:
         with:
           CACHE_VERSION: ${{ secrets.CACHE_VERSION }}
 
-      - name: Restore .next cache
+      - name: Restore Next.js build cache
         uses: actions/cache@v3
         with:
-          path: .next/cache
-          key: ${{ runner.os }}-next-cache-${{ github.sha }}
+          path: ${{ github.workspace }}/.next/cache
+          # Generate a new cache whenever packages or source files change.
+          key: ${{ runner.os }}-${{ secrets.CACHE_VERSION }}-nextjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+          # If source files changed but packages didn't, rebuild from a prior cache.
           restore-keys: |
-            ${{ runner.os }}-next-cache-${{ github.sha }}
-            ${{ runner.os }}-next-cache-
-
-      - name: Cache build
-        uses: actions/cache@v3
-        with:
-          path: .next
-          key: ${{ runner.os }}-next-build-${{ github.sha }}
```

Comment:
@znarf I didn't find any evidence that caching the entire `.next` folder has any benefits; NextJS docs recommend caching only `.next/cache`.